src/aat/java/uk/gov/hmcts/ccd/datastore/tests/functional/UpdateCaseTest.java

```diff
@@ -68,4 +68,188 @@ void shouldUpdateSingleField() {
              // Other fields not updated
              .body("TextField", equalTo(AATCaseBuilder.TEXT));
     }
+
+    @Test
+    @DisplayName("should update a case if the caseworker has 'CRUD' access on CaseType")
+    void shouldUpdateCaseWithCRUDAccessForCaseType() {
+        // Case Type with "CRUD" access to the role autoTestCaseWorker
+        final Long caseReference = Event.create("AAT_AUTH_15")
+            .as(asPrivateCaseworker(true))
+            .withData(FullCase.build())
+            .submitAndGetReference();
+
+        Event.update(caseReference)
+            .as(asAutoTestCaseworker())
+            .withData(
+                CaseData.builder()
+                    .numberField(UPDATED_NUMBER)
+                    .build()
+            )
+            .submit()
+
+            .then()
+            .statusCode(201)
+            .assertThat()
+            .rootPath("case_data")
+            .body("NumberField", equalTo(UPDATED_NUMBER))
+            .body("TextField", equalTo(AATCaseBuilder.TEXT));
+    }
+
+    @Test
+    @DisplayName("should update a case if the caseworker has 'U' access on CaseType")
+    void shouldUpdateCaseWithUAccessForCaseType() {
+        // Case Type with "U" access to the role autoTestCaseWorker
+        final Long caseReference = Event.create("AAT_AUTH_4")
+            .as(asPrivateCaseworker(true))
+            .withData(FullCase.build())
+            .submitAndGetReference();
+
+        Event.update(caseReference)
+            .as(asAutoTestCaseworker())
+            .withData(
+                CaseData.builder()
+                    .numberField(UPDATED_NUMBER)
+                    .build()
+            )
+            .submit()
+
+            .then()
+            .statusCode(201);
+    }
+
+    @Test
+    @DisplayName("should update a case if the caseworker has 'CU' access on CaseType")
+    void shouldUpdateCaseWithCUAccessForCaseType() {
+        // Case Type with "CU" access to the role autoTestCaseWorker
+        final Long caseReference = Event.create("AAT_AUTH_5")
+            .as(asPrivateCaseworker(true))
+            .withData(FullCase.build())
+            .submitAndGetReference();
+
+        Event.update(caseReference)
+            .as(asAutoTestCaseworker())
+            .withData(
+                CaseData.builder()
+                    .numberField(UPDATED_NUMBER)
+                    .build()
+            )
+            .submit()
+
+            .then()
+            .statusCode(201);
```

Comment:
When testing update, more fields should be updated and whether the values are indeed updated should be verified. For this, we should set a value to first X, verify that it's indeed X, then update it to Y, and verify that it's indeed Y. Otherwise, we may get a 201 response but may be not updating the fields. Ideally, we should verify this with all the fields we're can be updating with an update call.

Reply:
Mutlu,

  Only for few test methods where the user role doesn't have the "READ" privilege I didn't get the response content back as they only have the privilege to update the case but don't have read permission. Anyways, I have updated the these test methods, by fetching the response using the super user and asserted that the case data was successfully updated.